src/fontra/views/editor/editor.js

```diff
@@ -2030,6 +2104,259 @@ export class EditorController {
     return { contentElement, warningElement };
   }
 
+  selectionHasLockedGuidelines() {
+    const {
+      guidelineGlyph: guidelineGlyphSelection,
+      guidelineFont: guidelineFontSelection,
+    } = parseSelection(this.sceneController.selection);
+
+    const instance = this.sceneModel.getSelectedPositionedGlyph()?.glyph.instance;
+    for (const guidelineIndex of guidelineGlyphSelection || []) {
+      const guidelineGlyph = instance.guidelines[guidelineIndex];
+      if (guidelineGlyph.locked) {
+        return true;
+      }
+    }
+
+    //TODO: Guidelines Font
+    // check if any of the selected guidelines are locked
+
+    return false;
+  }
+
+  getLockGuidelineLabel(hasLockedGuidelines) {
+    const {
+      guidelineGlyph: guidelineGlyphSelection,
+      guidelineFont: guidelineFontSelection,
+    } = parseSelection(this.sceneController.selection);
+    const guidelinSelection = new Array().concat(
```

Comment:
Suggest to avoid contructing arrays that aren't really used apart from counting.

Maybe:
```js
    const numGuidelines = (guidelineGlyph?.length || 0) + (guidelineFont?.length || 0);
    // do things based on numGuidelines
```

Same for canLockGuideline() below.

Reply:
Thanks for the valid feedback. I changed it.